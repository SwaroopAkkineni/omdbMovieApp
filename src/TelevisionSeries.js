import React, {Component} from 'react';
import {Dimensions, FlatList, Platform, StyleSheet, Text, TouchableOpacity, View} from 'react-native';
import Season from './Season';
const { height, width } = Dimensions.get('window');

export default class TelevisionSeries extends React.Component {
  state = {
    fetchString: '',
    showName: '',
		seasons: 0,
	};

  componentDidMount() {
    this.processFetchString();
  }

  processFetchString = () => {
    const { navigation } = this.props;
    const showName = navigation.state.params.showName;
    let fetchString = 'http://www.omdbapi.com/?t=';

    showName.split(' ').forEach(function(str) {
      fetchString += str + '+'
    });
    fetchString += '&apikey=92e3aa84'
    console.log(fetchString);
    fetch(fetchString)
      .then((response) => response.json())
      .then((responseJson) => {
        console.log('responseJson: ', responseJson);
        this.setState({
          fetchString: fetchString,
          showName: showName,
          seasons: responseJson.totalSeasons,
        });
      })
      .catch((error) => {
        console.error(error);
      });
  }

  render() {
    const { showName, seasons } = this.state;
    const { navigate } = this.props.navigation;
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>{showName}</Text>
        <Text style={styles.welcome}>Season: {seasons}</Text>
        <FlatList
          data={Array.from({length: seasons}, (v, k) => k+1)}
          renderItem={({ item }) => (
            <TouchableOpacity style={styles.todoItem}
                  onPress={() => navigate('Season', { season: item, showName: showName})}>
              <Text style={styles.instructions}>Season {item} </Text>
            </TouchableOpacity>
          )}
        />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    textAlign: 'left',
    paddingLeft: width * 0.1,
    paddingRight: width * 0.1,
    backgroundColor: '#89D2DC',
  },
  todoItem: {
    alignItems: 'center',
    height: height * 0.075,
    borderBottomWidth: 1.5,
    borderColor: '#101D42',
    backgroundColor: '#89D2DC',
    flexDirection: 'row',
  },
  welcome: {
    color: '#101D42',
    fontWeight: 'bold',
    marginTop: width * 0.1,
    fontSize: height * 0.03,
  },
  instructions: {
    color: '#101D42',
  },
});
