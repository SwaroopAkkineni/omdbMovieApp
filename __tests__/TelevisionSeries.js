import React from 'react';
import 'react-native';
import TelevisionSeries from '../src/TelevisionSeries';

import renderer from 'react-test-renderer';
const navigationMock = { state: { params: { showName: "The Office" } } };

const mockResponse = (status, statusText, response) => {
        return new Response(response, {
            status: status,
            statusText: statusText,
            headers: {
                'Content-type': 'application/json'
            }
        });
    }; 

test('renders correctly', () => {
  fetch = jest.fn(() => new Promise(resolve => resolve()));
  const tree = renderer.create(<TelevisionSeries navigation={navigationMock}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
