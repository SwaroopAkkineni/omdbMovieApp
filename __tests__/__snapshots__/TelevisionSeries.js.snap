// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#89D2DC",
      "flex": 1,
      "paddingLeft": 75,
      "paddingRight": 75,
      "textAlign": "left",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#101D42",
        "fontSize": 40.019999999999996,
        "fontWeight": "bold",
        "marginTop": 75,
      }
    }
  >
    
  </Text>
  <Text
    style={
      Object {
        "color": "#101D42",
        "fontSize": 40.019999999999996,
        "fontWeight": "bold",
        "marginTop": 75,
      }
    }
  >
    Season: 
    0
  </Text>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
